go/vt/vtgate/evalengine/compiler_asm_push.go

```diff
@@ -279,6 +279,93 @@ func (asm *assembler) PushBVar_json(key string) {
 	}, "PUSH JSON(:%q)", key)
 }
 
+func push_datetime(env *ExpressionEnv, raw []byte) int {
+	env.vm.stack[env.vm.sp], env.vm.err = parseDateTime(raw)
+	env.vm.sp++
+	return 1
+}
+
+func (asm *assembler) PushColumn_datetime(offset int) {
```

Comment:
@vmg Do we treat `timestamp` columns the same as `datetime`?

Reply:
Yes, we've always done so in the evalengine. All the existing code does.